presto-main/src/main/java/com/facebook/presto/sql/SqlFormatterUtil.java

```diff
@@ -35,7 +37,8 @@ public static String getFormattedSql(Statement statement, SqlParser sqlParser, O
         // verify round-trip
         Statement parsed;
         try {
-            parsed = sqlParser.createStatement(sql);
+            ParsingOptions parsingOptions = new ParsingOptions(UNDEFINED /* formatted SQL should be unambiguous */);
```

Comment:
This only works because ExpressionFormatter produces DECIMAL 'xxx' for DecimalLiteral. I think that might be required for a while to avoid incompatibilities when creating views, for instance. Eventually, we need to change it to produce simple decimal literals. 

Add a TODO in ExpressionFormatter.visitDecimal to clean that up once the parse-as-double flag is removed.